iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/MandatoryLabelCheck.java

```diff
@@ -0,0 +1,120 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.checks;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.sonar.check.Rule;
+import org.sonar.check.RuleProperty;
+import org.sonar.iac.common.api.checks.CheckContext;
+import org.sonar.iac.common.api.checks.IacCheck;
+import org.sonar.iac.common.api.checks.InitContext;
+import org.sonar.iac.docker.symbols.ArgumentResolution;
+import org.sonar.iac.docker.tree.api.Body;
+import org.sonar.iac.docker.tree.api.DockerImage;
+import org.sonar.iac.docker.tree.api.LabelInstruction;
+import org.sonar.iac.docker.tree.api.OnBuildInstruction;
+
+@Rule(key = "S7028")
+public class MandatoryLabelCheck implements IacCheck {
+  private static final String MESSAGE = "Add the missing label%s: %s.";
+  private static final String DEFAULT_REQUIRED_LABELS = "maintainer";
+  private Set<String> requiredLabelSet = Set.of(DEFAULT_REQUIRED_LABELS);
+
+  @RuleProperty(
+    key = "labels",
+    description = "Comma separated list of required labels.",
+    defaultValue = DEFAULT_REQUIRED_LABELS)
+  String requiredLabels = DEFAULT_REQUIRED_LABELS;
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(Body.class, this::checkBody);
+    requiredLabelSet = splitLabels(requiredLabels);
+  }
+
+  static Set<String> splitLabels(String labels) {
+    return Arrays.stream(labels.split(","))
+      .map(String::trim)
+      .filter(s -> !s.isBlank())
+      .collect(Collectors.toSet());
+  }
+
+  private void checkBody(CheckContext ctx, Body body) {
+    Set<String> encounteredLabels = new HashSet<>();
+
+    body.dockerImages().stream()
+      .flatMap(MandatoryLabelCheck::gatherLabelInstructions)
+      .forEach(labelInstruction -> handleLabelInstruction(labelInstruction, encounteredLabels));
+
+    List<String> missingLabels = new ArrayList<>(requiredLabelSet);
+    missingLabels.removeAll(encounteredLabels);
+    if (!missingLabels.isEmpty()) {
+      if (missingLabels.size() == 1) {
+        ctx.reportIssue(body.dockerImages().get(0).from(), MESSAGE.formatted("", "\"" + missingLabels.get(0) + "\""));
```

Comment:
If a file doesn't have a `FROM`, i.e. contains only comments, the check for labels will fail, and here `get` will fail